File: 프로그래머스/LV0/코딩테스트 입문/햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    stack = [0] * len(ingredient)
    top = 0

    for i in range(len(ingredient)):
        stack[top] = ingredient[i]
        top += 1

        if top >= 3 and stack[top-4:top] == [1,2,3,1]:
            answer += 1
            top -= 4
            
    return answer

File: 프로그래머스/LV1/코딩테스트 연습/햄버거 만들기/햄버거 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// ingredient_len은 배열 ingredient의 길이입니다.
int solution(int ingredient[], size_t ingredient_len)
{
    int answer = 0;
    int top = 0;
    char compare[4] = { 1,2,3,1 };
    char* stack = (char*)malloc(ingredient_len);
    
    for (int i = 0; i < ingredient_len; i++)
    {
        stack[top] = (char)ingredient[i];
        top++;

        if (top >= 3 && !memcmp(&stack[top - 4], compare, 4))
        {
            answer++;
            top -= 4;
        }
    }
    return answer;
}